LeafBot/LeafBot/Program.cs

```diff
@@ -18,13 +18,21 @@ class Program
   {
     public const string VERSION = "0.5";
 
+    public static string BasePath { get; private set; }
+    public static string DataPath { get; private set; }
+
     public DiscordClient Client { get; set; }
     public CommandsNextExtension Commands { get; set; }
     private Timer SaveTimer;
-    private static string configPath = "config.json";
+    private static string configPath;
 
     static void Main(string[ ] args)
     {
+      // Setup all our bot's paths
+      BasePath = AppDomain.CurrentDomain.BaseDirectory;
+      DataPath = Path.Combine(BasePath, "Data");
+      configPath = Path.Combine(BasePath, "config.json");
```

Comment:
config is now pointing to `LeafBot\\LeafBot\\bin\\Debug\\netcoreapp3.1\\config.json`? we def wanna move the config file here?

Reply:
It will basically always try and look for the config.json in the same directory as the assembly itself (which will be that path for me while in vs studio). I'm alright with it looking in the same directory but we can move it to a subforlder (a bit like Data) if you want. Hopefully this change won't require us to shuffle and source files around 🤞

Reply:
ok so you're proposing to move config into there? imo I think the config file should be in a more obvious location like where it is rn now, no? it's the same location that you put the config in potatobot - how were you resolving it's location for that?

Reply:
Yeah just to clarify this won't actually change the location of config.json. It will still need to be in the same location as before (in the same directory as the bot assembly). This just changes it from being referenced using a relative path to and an explicit path (```AppDomain.CurrentDomain.BaseDirectory``` gives us the path the assembly is in).

I agree with ya the config file should be somewhere obvious so I think where it is at the moment is a good place, it's also where I had it with potatobot (I think I was just referring to it like we were before with a relative path :/)

Reply:
resolved confusion over chat. see PR approval comment for summary